assignment-1/Functions.java

```diff
@@ -0,0 +1,271 @@
+import java.io.*;
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static java.util.stream.Collectors.toMap;
+
+public class Functions {
+
+    public static final String delimiter = ",";
+    public static ArrayList<Players> allPlayers = new ArrayList<Players>();
+    public static ArrayList<Team> allTeams = new ArrayList<Team>();
+    public static <Player> void readData(String csvFile) {
+        try {
+            File file = new File(csvFile);
+            FileReader fr = new FileReader(file);
+            BufferedReader br = new BufferedReader(fr);
+            String line = "";
+            String[] individualField;
+
+            //allPlayers
+
+            while ((line = br.readLine()) != null) {
+                while ((line = br.readLine()) != null) {
+                    individualField = line.split(delimiter);
+                    String playerName = individualField[0];
+                    String playerTeam = individualField[1];
+                    String playerRole = individualField[2];
+                    int playerMatches = Integer.parseInt(individualField[3]);
+                    int playerRuns = Integer.parseInt(individualField[4]);
+                    Double playerAverage = Double.parseDouble(individualField[5]);
+                    Double playerStrikeRate = Double.parseDouble(individualField[6]);
```

Comment:
This is susceptible to arrayIndexOutOfBoundException
Should have handled this edge case